src/couch/src/couch_server.erl

```diff
@@ -219,6 +219,7 @@ close_db_if_idle(DbName) ->
 
 init([]) ->
     couch_util:set_mqd_off_heap(?MODULE),
+    process_flag(priority, high),
```

Comment:
Slightly worried about this one. Say if couch server is looking through the LRU to make room or is running regexes in delete it might completely block any other processes, especially  if there is just one scheduler available. Maybe that's desirable to block incoming requests, but I can also see it being unpredictable.

Could make it conditional on the number of schedulers present or a config value?

Reply:
Not a bad idea to make it conditional on the number of schedulers. Granted if there weren't more than one scheduler I'd be fairly surprised that the node would be able to sustain enough load elsewhere to cause this to be a bottleneck.

Reply:
Fair point on the single scheduler case.

We have the `couch_util:set_mqd_off_heap(?MODULE)` pattern, we could re-use that for setting priorities too?Thinking about possibly getting unexpected behavior, since we haven't been playing with process priority before, it might be helpful to have a way to turn it off in config.

Reply:
Done.